src/plugins/data_source/server/saved_objects/credential_saved_objects_client_wrapper.ts

```diff
@@ -138,15 +138,27 @@ export class CredentialSavedObjectsClientWrapper {
   }
 
   private async validateAndEncryptPartialAttributes<T = unknown>(attributes: T) {
-    this.validateCredentialMaterials(attributes.credentialMaterials);
+    const { credentialMaterials } = attributes;
+    const { credentialMaterialsContent } = credentialMaterials;
+
+    if ('password' in credentialMaterialsContent) {
+      this.validatePassword(credentialMaterialsContent.password);
+      return {
+        ...attributes,
+        credentialMaterials: await this.encryptUsernamePasswordTypedCredentialMaterials(
+          credentialMaterials
+        ),
+      };
+    } else {
+      this.validateAttributes(attributes);
+    }
 
     return await this.encryptCredentialMaterials(attributes);
   }
 
   private validateAttributes<T = unknown>(attributes: T) {
     const { title, credentialMaterials } = attributes;
-
-    if (title === undefined) {
+    if (!title) {
```

Comment:
Why only change validation on title to undefined?

Reply:
Since when title is an empty string, it will not have error if use `title == undefined`